index.js

```diff
@@ -516,25 +550,26 @@ class ClientServiceBase {
   }
 
   /**
+   * @param {ContractExecutionRequest} request
    * @param {ContractExecutionResponse} response
    * @return {Boolean} return false if Auditor is enabled and the results
    * from Ledger and Auditor don't match
    */
-  async _executeInAuditor(response) {
+  async _executeValidation(request, response) {
     if (!this._isAuditorEnabled()) {
       return true;
     }
     const promise = new Promise((resolve, reject) => {
-      this.auditorClient.executeContractWithProofs(
-          this._createContractExecutionRequestWithAssetProofs(response),
+      this.auditorClient.validateExecution(
+          this._createExecutionValidationRequest(request, response),
           this.metadata,
           (err, auditorResponse) => {
             if (err) {
               reject(err);
             } else {
               const auditorResult =
                 ContractExecutionResult.fromGrpcContractExecutionResponse(
-                  auditorResponse,
+                    auditorResponse,
                 );
               if (auditorResult.getProofs().length > 0) {
```

Comment:
@feeblefakie Thank you for reviewing. Ahh, in such cases, it will fall to `resolve(true)`...  I will fix it. Do all the non-malicious responses have at least one proof in any case? Just a confirmation.

Reply:
@feeblefakie 444be7c will fix the above issue if non-malicious responses always have at least one proof. PTAL. Thanks!

Reply:
> Do all the non-malicious responses have at least one proof in any case?

Maybe yes but there seem some corner cases like if a contract doesn't do anything since something else is not ready?
Maybe checking both results if they are consistent is better.
(if one's proof is empty then the other's proof has to be empty,
if one's proof is non-empty then the other's proof has to be non-empty, and they have to be consistent)

Reply:
@feeblefakie 
I revise the code snippet here
https://github.com/scalar-labs/scalardl-javascript-sdk-base/pull/57/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R607

Now, if either auditor or ledger return empty proofs but one of them has non-empty proofs, the validation will be determined inconsistent.

Reply:
@supl Thank you! Let me check.